test/diff-highlighter-2.html

```
@@ -0,0 +1,78 @@
+    <template>
+      <diff-highlighter>
+      </diff-highlighter>
+    </template>
+  </test-fixture>
+  <script>
+
+     suite('<diff-highlighter>', function() {
+      var elem;
+      setup(function() {
+        elem = fixture('basic');
+        elem.code = 'display: inline-block;\n        padding: 3px 6px;\n        border-radius: 6px;\n-       color: var(--primary-background-color);\n-       color: var(--tag-text-color);\n-       background-color: var(--pr-status-color);\n-       background-color: var(--tag-color);\n+       color: var(--primary-background-color, --tag-text-color);\n+       background-color: var(--pr-status-color, --tag-color);\n+       overflow: hidden;\n+       text-overflow: ellipsis;\n+       max-width: 120%;\n      }\n      iron-icon {\n        vertical-align: text-bottom;"';
+        elem.beforeStart = 0;
+        elem.afterStart = 1;
+
+      });
+      test('tap button fires event', function() {
+        elem.addEventListener('split', function(e) {
+          assert.equal(e.detail.lineNumber, 1);
+        });
+        var icon = Polymer.dom(elem.root).querySelector('iron-icon');
+        MockInteractions.tap(icon);
+      });
+
+     test('tapping all buttons, fires events', function(done) {
+        var count = 0;
+        elem.addEventListener('split', function() {
+          count++;
+          if (count === 14) {
+            done();
+          }
+        });
+        var icons = Polymer.dom(elem.root).querySelectorAll('iron-icon');
+        flush(function() {
+          icons.forEach(function(icon) {
+            MockInteractions.tap(icon);
+          });
+        });
+      });
+
+    test('tapping on last line should not fire', function(done) {
+        elem.code = 'one liner \n'
+        var icon = Polymer.dom(elem.root).querySelector('iron-icon');
+         flush(function() {
+          elem.addEventListener('split', function() {
+            done(new Error('This event should not have been fired'));
```

You can remove the `done` from this line

---

And can you define the event listener before the call to `flush`? This would be more consistent with the rest of our tests

---

If get this call the test should fail. 
[I read that making a done with text would let the test fail ](http://stackoverflow.com/a/40539753/4760422).  I can also do `assert.equal(1,0)` but I thought this would be better.

---

I see, personally I've always done `throw new Error(...)` to let a test fail but I guess this is fine too